File: Javascript/3856.trim_trailing_vowels [1].js
/**
 * @param {string} s
 * @return {string}
 */
var trimTrailingVowels = function(s) {
    function isVowel(ch){
        return ch === 'a' || ch === 'e' || ch === 'i' || ch === 'o' || ch === 'u'
    }

    const chars = s.split('')
    
    for(let i=chars.length-1; i>=0; i--){
        if(isVowel(chars[i]))
            chars.pop()
        else
            break
    }

    return chars.join('')
};

File: CPP [3501-4000]/3856.trim-trailing-vowels [1].cpp
class Solution {
private:
    bool isVowel(const char ch){
        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
    }

public:
    // T.C.=O(n), S.C.=O(1)
    string trimTrailingVowels(string s) {
        for(int i=s.length()-1; i>=0; i--){
            if(isVowel(s[i]))
                s.pop_back();
            else
                break;
        }
        
        return s;
    }
};